sound/soc/sof/ipc4-pcm.c

```diff
@@ -299,7 +299,7 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	pipeline_list = &spcm->stream[substream->stream].pipeline_list;
 
 	/* nothing to trigger if the list is empty */
-	if (!pipeline_list->pipelines)
+	if (!pipeline_list->pipelines || !pipeline_list->count)
```

Comment:
@ujfalusi is this correct? count is only incremented in https://github.com/thesofproject/linux/blob/d39118493a36bb285c10fd763aa3f0954957b46a/sound/soc/sof/ipc4-pcm.c#L65. So it will always be 0 at first no?

Reply:
it is incremented in sof_set_up_widgets_in_path():
https://github.com/thesofproject/linux/blob/5694a58bb1e702c05b43134021d382df53529735/sound/soc/sof/sof-audio.c#L465

it is only going to be zero if the sof_widget_setup() earlier fail for the first widget
https://github.com/thesofproject/linux/blob/5694a58bb1e702c05b43134021d382df53529735/sound/soc/sof/sof-audio.c#L445

Reply:
ahh yes sorry I confused it with the trigget_list->count. This isnt the same